Getting [lotus-worker] Error 1 when installing

Environment: 2021 MacBook Pro (12.0.1), ARM architecture.
Homebrew has been updated and upgrated.

When I run the `brew install lotus` command, I get:
```
Already downloaded: /Users/josef1/Library/Caches/Homebrew/downloads/9c4142aad51c88bc3c10b00efb944ff0f3b886a8b393d9f7c5632f9b5a2adf1c--lotus-1.13.0.tar.gz
==> Installing lotus from filecoin-project/lotus
==> git clone https://github.com/filecoin-project/filecoin-ffi.git extern/filecoin-ffi
==> git clone https://github.com/filecoin-project/serialization-vectors.git extern/serialization-vectors
==> git clone https://github.com/filecoin-project/test-vectors.git extern/test-vectors
==> touch build/.update-modules
==> make all
Last 15 lines from /Users/josef1/Library/Logs/Homebrew/lotus/05.make:
go: downloading github.com/ipfs/go-filestore v1.0.0
go: downloading github.com/ipfs/go-ipfs-chunker v0.0.5
go: github.com/filecoin-project/filecoin-ffi@v0.30.4-0.20200910194244-f640612a1a1f requires
	github.com/filecoin-project/go-state-types@v0.1.1: missing go.sum entry; to add it:
	go mod download github.com/filecoin-project/go-state-types
go: github.com/filecoin-project/filecoin-ffi@v0.30.4-0.20200910194244-f640612a1a1f requires
	github.com/filecoin-project/go-state-types@v0.1.1: missing go.sum entry; to add it:
	go mod download github.com/filecoin-project/go-state-types
go: github.com/filecoin-project/filecoin-ffi@v0.30.4-0.20200910194244-f640612a1a1f requires
	github.com/filecoin-project/go-state-types@v0.1.1: missing go.sum entry; to add it:
	go mod download github.com/filecoin-project/go-state-types
make: *** [lotus-worker] Error 1
make: *** Waiting for unfinished jobs....
make: *** [lotus-miner] Error 1
make: *** [lotus] Error 1

If reporting this issue please do so at (not Homebrew/brew or Homebrew/core):
  https://github.com/filecoin-project/homebrew-lotus/issues
```

Any ideas what I might be missing?

--- a/Formula/lotus.rb
+++ b/Formula/lotus.rb
@@ -9,14 +9,57 @@ class Lotus < Formula
   depends_on "pkg-config" => :build
   depends_on "go"         => :build
   depends_on "jq"         => :build
+  depends_on "bzr"        => :build
+  depends_on "rustup-init" => :build
   depends_on "hwloc"
 
-  def install
-    # update submodules
-    system "git clone https://github.com/filecoin-project/filecoin-ffi.git extern/filecoin-ffi"
-    system "git clone https://github.com/filecoin-project/serialization-vectors.git extern/serialization-vectors"
-    system "git clone https://github.com/filecoin-project/test-vectors.git extern/test-vectors"
+  def arch_M1
+    IO.popen(['arch']).read.strip === 'arm64'
+  end
+
+  def install_submodules
+    system "git clone --recurse-submodules https://github.com/filecoin-project/filecoin-ffi.git extern/filecoin-ffi"
+    system "git clone --recurse-submodules https://github.com/filecoin-project/serialization-vectors.git extern/serialization-vectors"
+    system "git clone --recurse-submodules https://github.com/filecoin-project/test-vectors.git extern/test-vectors"
     system "touch build/.update-modules"
+  end
+
+  def build_filecoin_ffi
+    ## First install rust to prevent Error on filecoin-ffi build:
+    ##    ++ command -v rustup
+    ##    + '[' -x '' ']'
+    ##    + echo '[build_from_source] Error: rustup is not installed.'
+    ##    [build_from_source] Error: rustup is not installed.
+    ##    + echo '[build_from_source] install Rust toolchain installer to resolve this problem.'
+    ##    [build_from_source] install Rust toolchain installer to resolve this problem.
+    ##    + exit 1
+
+    # Set cargo_home to local directory
+    system 'mkdir -p extern/cargo_home'
+    ENV['CARGO_HOME'] = 'extern/cargo_home'
+    system 'rustup-init -y'
+
+    # Set environment variables so M1 mac
+    ENV['LIBRARY_PATH'] = "#{ENV['HOMEBREW_PREFIX']}/lib"
+    ENV['FFI_BUILD_FROM_SOURCE'] = '1'
+    cargo_bin = "#{IO.popen(['pwd']).read.strip}/extern/cargo_home/bin"
+    ENV['PATH'] += ":#{cargo_bin}"
+    puts "PATH = #{ENV['PATH']}"
+
+    # Build filecoin-ffi from source for Mac M1
+    system 'cd extern/filecoin-ffi && make'
+
+
+    system 'go mod download github.com/filecoin-project/go-state-types'
+    system 'go get github.com/filecoin-project/filecoin-ffi@v0.30.4-0.20200910194244-f640612a1a1f'
+  end
+
+  def install
+
+    # update submodules
+    install_submodules
+
+    build_filecoin_ffi if arch_M1
 
     # make
     system "make all"

